src/mgr/PyFormatter.h

```diff
@@ -30,6 +30,8 @@
 class PyFormatter : public ceph::Formatter
 {
 public:
+  PyFormatter (const PyFormatter&) = delete;
+  PyFormatter& operator= (const PyFormatter&) = delete;
```

Comment:
i don't understand this change. if a copy ctor is not supposed to be called, we can either mark it `delete` in C++11 or make it a `private` member in C++98. with your change, the copy ctor will be implicitly created by the compiler, and moreover, we have no way to prevent developer to copy a `PyFormatter` instance intentionally or unintentioally.

Reply:
It was my understanding that this does exactly mark the copy ctor as `delete`.

Reply:
i strongly doubt it.

Reply:
Trying to use PyFormatter's copy constructor with this change throws a compile time error:

```
/home/jan/code/ceph/ceph/src/mgr/ActivePyModules.cc: In member function ‘PyObject* ActivePyModules::with_perf_counters(std::function<void(PerfCounterInstance&, PerfCounterType&, PyFormatter&)>, const string&, const string&, const string&) const’:
/home/jan/code/ceph/ceph/src/mgr/ActivePyModules.cc:617:19: error: use of deleted function ‘PyFormatter::PyFormatter(const PyFormatter&)’
   PyFormatter f2(f);
                   ^
In file included from /home/jan/code/ceph/ceph/src/mgr/ActivePyModules.cc:20:
/home/jan/code/ceph/ceph/src/mgr/PyFormatter.h:33:3: note: declared here
   PyFormatter (const PyFormatter&) = delete;
   ^~~~~~~~~~~
```

Reply:
@jan--f thanks for testing it. i must have read something different or just misread it. i thought you were deleting the 
```c++
PyFormatter (const PyFormatter&) = delete;
PyFormatter& operator= (const PyFormatter&) = delete;
```
from this source file.

sorry for my mistake. =(

Reply:
no worries @tchaikov :)

Reply:
Yes indeed, it's one of (though certainly not the only) canonical usages, discussed in the original proposal (N2210):
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2007/n2210.html
A nice discussion may be found here:
https://abseil.io/tips/143